datastore/longhorn.go

```diff
@@ -967,8 +967,28 @@ func (s *DataStore) CheckEngineImageReadiness(image string, nodes ...string) (is
 	return true, nil
 }
 
-// CheckEngineImageReadinessForVolume checks if the IMAGE is deployed on the NODEID as well as all the volume's replicas
-func (s *DataStore) CheckEngineImageReadinessForVolume(image, volumeName, nodeID string) (isReady bool, err error) {
+// CheckEngineImageReadyOnAtLeastOneVolumeReplica checks if the IMAGE is deployed on the NODEID and on at least one of the the volume's replicas
+func (s *DataStore) CheckEngineImageReadyOnAtLeastOneVolumeReplica(image, volumeName, nodeID string) (isReady bool, err error) {
+	replicas, err := s.ListVolumeReplicas(volumeName)
+	if err != nil {
+		return false, fmt.Errorf("cannot get replicas for volume %v: %w", volumeName, err)
+	}
+
+	isReady, err = s.CheckEngineImageReadiness(image, nodeID)
+	if nodeID != "" && !isReady {
+		return
```

Comment:
I feel like it's better to specify the return values explicitly instead of relying on the default values of the named params. In this case you also probably don't need the named params, since it's obvious from the usage :)

Reply:
Thank you. Fixed